core/geotime/src/main/java/org/locationtech/geowave/core/geotime/store/query/aggregate/GeohashBinningStrategy.java

```diff
@@ -0,0 +1,136 @@
+package org.locationtech.geowave.core.geotime.store.query.aggregate;
+
+import org.locationtech.geowave.core.geotime.store.dimension.GeometryWrapper;
+import org.locationtech.geowave.core.store.data.CommonIndexedPersistenceEncoding;
+import org.locationtech.geowave.core.store.data.PersistentDataset;
+import org.locationtech.geowave.core.store.index.CommonIndexValue;
+import org.locationtech.geowave.core.store.query.aggregate.AggregationBinningStrategy;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.geom.Point;
+import org.opengis.feature.simple.SimpleFeature;
+
+public class GeohashBinningStrategy<T> implements AggregationBinningStrategy<T> {
+
+  private int precision;
+
+  /**
+   * Create a Binning Strategy based on GeoHashes.
+   *
+   * @param maxBinsAllowed TODO: explain how this invariant works, and how itll break for < 32.
+   */
+  public GeohashBinningStrategy(int maxBinsAllowed) {
+    this.precision = getDesiredPrecision(maxBinsAllowed);
+  }
+
+  /**
+   * Given the maximum bins allowed, figure out a Geohash precision that is less than or equal to
+   * that number.
+   *
+   * @param maxBins The max number of bins to allow during processing. For example, given
+   *        {@code maxBins = 33}, a precision of 1 is given, because a precision of 2 can contain
+   *        1024 hashes ({@code 32^2}), which is too many. Given {@code maxBins = 30}, a precision
+   *        of 1 is given, because 1 is the smallest possible. Given {@code maxBins > 32^12}, 12 is
+   *        returned, because 12 is the max precision.
+   * @return A geohash precision that, in the worst case, would contain {@code maxBins} different
+   *         hashes.
+   */
+  private static int getDesiredPrecision(long maxBins) {
+    for (int i = 12; i > 0; i--) {
+      if ((double) maxBins >= Math.pow(32, i)) {
+        return i;
+      }
+    }
+    // TODO: should this be 0, AKA 1 bin for all data? (if so make sure to fix tests!)
```

Comment:
Making this 0 would be the same as not binning, should that be the behavior here?

Reply:
Resolved through on of JD's comments